src/Microsoft.Extensions.Configuration.AzureAppConfiguration/IConfigurationRefresher.cs

```diff
@@ -21,6 +21,7 @@ public interface IConfigurationRefresher
         /// The refresh operation failed with one or more errors. Check <see cref="AggregateException.InnerExceptions"/> for more details.
         /// </exception>
         /// <exception cref="InvalidOperationException">The refresh operation was invoked before Azure App Configuration Provider was initialized.</exception>
+        /// <exception>An Azure.Identity.AuthenticationFailedException is thrown if authentication using TokenCredential passed to <see cref="AzureAppConfigurationKeyVaultOptions.SetCredential"/> method failed.</exception>
```

Comment:
We can't document this we don't own it.

Reply:
Not sure if there is a better way to do this. I understand we don't reference the package that has this exception type. With this, customers can handle it if they want at least. The `AuthenticationFailedException` should really be from either Key Vault client or Azure.Core. We should open an issue against the SDK.

Reply:
@jimmyca15 We document this exception since there are valid scenarios that will cause `RefreshAsync` method to throw it and we expect the users to handle it. I know this is not ideal, since we don't own it. It still seems to be better than crashing user's application even when a user handles all the documented exceptions.

Reply:
@zhenlan I already have issue https://github.com/Azure/azure-sdk-for-net/issues/10759 open against the SDK team. It seems they are considering moving the exception to `Azure.Core` similar to what we were discussing offline. However, I don't expect that change to be available before our next release.

Reply:
@abhilasharora documenting the exception like this is similar to taking a dependency on it. If we feel the need to do that perhaps we should just catch it like we are doing right now in TryRefreshAsync and wrap it in a KeyVaultReference Exception.